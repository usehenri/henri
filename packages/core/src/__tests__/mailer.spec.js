const BaseModule = require('../base/module');
const Henri = require('../henri');
const Mailer = require('../1.mailer');

const fetch = require('isomorphic-fetch');

describe('mailer', () => {
  beforeAll(async () => {
    this.henri = new Henri({ runlevel: 1 });
    this.henri.forceMail = true;
    await this.henri.init();
  });

  afterAll(async () => {
    await this.henri.stop();
  });

  test('should be defined', () => {
    expect(this.henri.mail).toBeDefined();
  });

  test('should extend BaseModule', () => {
    expect(this.henri.mail).toBeInstanceOf(BaseModule);
  });

  test('should match snapshot', () => {
    const mailer = new Mailer();

    expect(mailer).toMatchSnapshot();
  });

  test('should populate testAccount', () => {
    expect(this.henri.mail.testAccount).toBeTruthy();
  });

  test('should not send malformed emails', async () => {
    await expect(this.henri.mail.send()).rejects.toBeDefined();
  });

  test('should send mails', async () => {
    const info = await this.henri.mail.send({
      from: 'robot@usehenri.io',
      subject: 'Hey there',
      text: 'Whats up?',
      to: 'felix@usehenri.io',
    });
    const url = this.henri.mail.nodemailer.getTestMessageUrl(info);

    expect(info).toBeTruthy();
    await expect(fetch(url).then(res => res.status)).resolves.toEqual(200);
  });

  test('should not send if transporter is not defined', async () => {
    delete this.henri.mail.transporter;

    expect(this.henri.mail.transporter).toBeUndefined();

    await expect(this.henri.mail.send()).rejects.toBeDefined();
  });

  test('should throw if no config and not testing', () => {
    this.henri.isTest = false;
    delete this.henri.mail.transporter;
    this.henri.mail.init();
    expect(this.henri.mail.transporter).toBeUndefined();
    this.henri.isTest = true;
  });
});
