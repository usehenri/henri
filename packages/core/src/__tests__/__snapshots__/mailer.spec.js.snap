// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mailer should match snapshot 1`] = `
Mailer {
  "config": undefined,
  "consoleOnly": false,
  "henri": undefined,
  "info": [Function],
  "init": [Function],
  "key": null,
  "name": "mail",
  "nodemailer": Object {
    "createTestAccount": [Function],
    "createTransport": [Function],
    "getTestMessageUrl": [Function],
  },
  "reloadable": false,
  "runlevel": 1,
  "send": [Function],
  "setup": [Function],
  "start": [Function],
  "testAccount": undefined,
  "transporter": undefined,
}
`;
